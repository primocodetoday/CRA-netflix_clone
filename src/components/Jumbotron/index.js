import React from 'react';
import PropTypes from 'prop-types';
import {
  Item,
  Inner,
  Container,
  Pane,
  Title,
  SubTitle,
  Image,
} from './styles/jumbotron';

const Jumbotron = ({ children, direction }, ...restProps) => {
  return (
    <Item {...restProps} direction={direction}>
      <Inner>{children}</Inner>
    </Item>
  );
};

Jumbotron.Container = function JumbotronContainer({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
};

Jumbotron.Pane = function JumbotronPane({ children, ...restProps }) {
  return <Pane {...restProps}>{children}</Pane>;
};

Jumbotron.Title = function JumbotronTitle({ children, ...restProps }) {
  return <Title {...restProps}>{children}</Title>;
};

Jumbotron.SubTitle = function JumboSubtronTitle({ children, ...restProps }) {
  return <SubTitle {...restProps}>{children}</SubTitle>;
};

Jumbotron.Image = function JumbotronImage({ ...restProps }) {
  return <Image {...restProps} />;
};

Jumbotron.propTypes = {
  direction: PropTypes.string,
};

Jumbotron.defaultProps = {
  direction: 'row',
};

export default Jumbotron;
