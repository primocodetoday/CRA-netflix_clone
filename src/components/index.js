export { default as Footer } from 'components/Footer';
export { default as Accordion } from 'components/Accordion';
export { default as Jumbotron } from 'components/Jumbotron';
