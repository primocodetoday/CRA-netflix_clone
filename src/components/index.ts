export { Accordion } from 'components/Accordion/Accordion';
export { Banner } from 'components/Banner/Banner';
export { Card } from 'components/Card/Card';
export { CTAForm } from 'components/CTAForm/CTAForm';
export { Form } from 'components/Form/Form';
export { Footer } from 'components/Footer/Footer';
export { Header } from 'components/Header/Header';
export { Jumbotron } from 'components/Jumbotron/Jumbotron';
export { Loading } from 'components/Loading/Loading';
export { Player } from 'components/Player/Player';
export { Profiles } from 'components/Profiles/Profiles';
