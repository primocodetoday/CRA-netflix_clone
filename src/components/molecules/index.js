export { default as Footer } from 'components/molecules/Footer/Footer';
export { default as Accordion } from 'components/molecules/Accordion/Accordion';
export { default as Jumbotron } from 'components/molecules/Jumbotron/Jumbotron';
export { default as CTAForm } from 'components/molecules/CTAForm/CTAForm';
export { default as Header } from 'components/molecules/Header/Header';
