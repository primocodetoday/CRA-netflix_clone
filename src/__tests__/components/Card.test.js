import React from 'react';
import { render, fireEvent } from '@testing-library/react';

import { Card, Player } from 'components';

const category = 'series';
const slideRows = [
  {
    title: 'Documentaries',
    data: [
      {
        genre: 'documentaries',
        maturity: '18',
        slug: 'tiger-king',
        description: 'Tiger King description',
        id: '07adcf19-517f-41b4-8f8b-04fee694bce1',
        title: 'Tiger King',
        docId: 'IcRxJBanzE7nmIp2GMGb',
      },
    ],
  },
  {
    title: 'Feel Good',
    data: [
      {
        title: 'Juno',
        genre: 'feel-good',
        description: 'Juno description',
        maturity: '0',
        id: 'a938b5a6-fe25-4a06-8050-353bc7146ccd',
        slug: 'juno',
        docId: '4JDgdf5vE0K3YrW9ZnbP',
      },
    ],
  },
];

describe('<Card/> renders properly', () => {
  it('with data', () => {
    const { container, getByText } = render(
      <Card.Row>
        {slideRows.map((slideItem) => (
          <Card key={`${category}-${slideItem.title.toLowerCase()}`}>
            <Card.Title>{slideItem.title}</Card.Title>
            <Card.Entities>
              {slideItem.data.map((item) => (
                <Card.Item key={item.docId} item={item}>
                  <Card.Poster
                    src={`/assets/images/${category}/${item.genre}/${item.slug}/small.jpg`}
                  />
                  <Card.Meta>
                    <Card.SubTitle>{item.title}</Card.SubTitle>
                    <Card.Text>{item.description}</Card.Text>
                  </Card.Meta>
                </Card.Item>
              ))}
            </Card.Entities>
            <Card.Feature category={category}>
              <Player>
                <Player>
                  <Player.Button />
                  <Player.Video src="assets/videos/jellyfish.mp4" />
                </Player>
              </Player>
            </Card.Feature>
          </Card>
        ))}
      </Card.Row>,
    );

    expect(getByText('Documentaries')).toBeTruthy();
    expect(getByText('Tiger King')).toBeTruthy();
    expect(getByText('Tiger King description')).toBeTruthy();

    expect(getByText('Feel Good')).toBeTruthy();
    expect(getByText('Juno')).toBeTruthy();
    expect(getByText('Juno description')).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('with data and toggle feature', () => {
    const { container, queryByText, getByTestId, getByAltText } = render(
      <Card.Row>
        {slideRows.map((slideItem) => (
          <Card key={`${category}-${slideItem.title.toLowerCase()}`}>
            <Card.Title>{slideItem.title}</Card.Title>
            <Card.Entities>
              {slideItem.data.map((item) => (
                <Card.Item
                  key={item.docId}
                  item={item}
                  data-testid={`${item.slug}-feature`}
                >
                  <Card.Poster
                    src={`/assets/images/${category}/${item.genre}/${item.slug}/small.jpg`}
                  />
                  <Card.Meta>
                    <Card.SubTitle>{item.title}</Card.SubTitle>
                    <Card.Text>{item.description}</Card.Text>
                  </Card.Meta>
                </Card.Item>
              ))}
            </Card.Entities>
            <Card.Feature category={category}>
              <Player>
                <Player>
                  <Player.Button />
                  <Player.Video src="assets/videos/jellyfish.mp4" />
                </Player>
              </Player>
            </Card.Feature>
          </Card>
        ))}
      </Card.Row>,
    );

    expect(queryByText('18')).toBeFalsy();

    fireEvent.click(getByTestId('tiger-king-feature'));

    expect(queryByText('18')).toBeTruthy();
    fireEvent.click(getByAltText('close'));
    expect(queryByText('18')).toBeFalsy();

    expect(queryByText('PG')).toBeFalsy();
    fireEvent.click(getByTestId('juno-feature'));
    expect(queryByText('PG')).toBeTruthy();
    fireEvent.click(getByAltText('close'));
    expect(queryByText('PG')).toBeFalsy();

    expect(container.firstChild).toMatchSnapshot();
  });
});
