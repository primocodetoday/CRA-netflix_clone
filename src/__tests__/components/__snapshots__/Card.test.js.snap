// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders properly Renders Card and toggle feature 1`] = `
<div
  class="StyledCard__Row-sc-5awhqw-3 eEqxOc"
>
  <div
    class="StyledCard__Container-sc-5awhqw-1 FaQVJ"
  >
    <h1
      class="StyledCard__Title-sc-5awhqw-0 fFvuDs"
    >
      Documentaries
    </h1>
    <div
      class="StyledCard__Entities-sc-5awhqw-12 jAzA-dB"
    >
      <div
        class="StyledCard__Item-sc-5awhqw-13 hIuTEY"
        data-testid="tiger-king-feature"
      >
        <img
          class="StyledCard__Poster-sc-5awhqw-14 hGbFEx"
          src="/assets/images/series/documentaries/tiger-king/small.jpg"
        />
        <div
          class="StyledCard__Meta-sc-5awhqw-11 cwrXet"
        >
          <p
            class="StyledCard__SubTitle-sc-5awhqw-2 bNAbEZ"
          >
            Tiger King
          </p>
          <p
            class="StyledCard__Text-sc-5awhqw-4 bLrHhJ"
          >
            Tiger King description
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    class="StyledCard__Container-sc-5awhqw-1 FaQVJ"
  >
    <h1
      class="StyledCard__Title-sc-5awhqw-0 fFvuDs"
    >
      Feel Good
    </h1>
    <div
      class="StyledCard__Entities-sc-5awhqw-12 jAzA-dB"
    >
      <div
        class="StyledCard__Item-sc-5awhqw-13 hIuTEY"
        data-testid="juno-feature"
      >
        <img
          class="StyledCard__Poster-sc-5awhqw-14 hGbFEx"
          src="/assets/images/series/feel-good/juno/small.jpg"
        />
        <div
          class="StyledCard__Meta-sc-5awhqw-11 cwrXet"
        >
          <p
            class="StyledCard__SubTitle-sc-5awhqw-2 bNAbEZ"
          >
            Juno
          </p>
          <p
            class="StyledCard__Text-sc-5awhqw-4 bLrHhJ"
          >
            Juno description
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Card renders properly Renders Card with data 1`] = `
<div
  class="StyledCard__Row-sc-5awhqw-3 eEqxOc"
>
  <div
    class="StyledCard__Container-sc-5awhqw-1 FaQVJ"
  >
    <h1
      class="StyledCard__Title-sc-5awhqw-0 fFvuDs"
    >
      Documentaries
    </h1>
    <div
      class="StyledCard__Entities-sc-5awhqw-12 jAzA-dB"
    >
      <div
        class="StyledCard__Item-sc-5awhqw-13 hIuTEY"
      >
        <img
          class="StyledCard__Poster-sc-5awhqw-14 hGbFEx"
          src="/assets/images/series/documentaries/tiger-king/small.jpg"
        />
        <div
          class="StyledCard__Meta-sc-5awhqw-11 cwrXet"
        >
          <p
            class="StyledCard__SubTitle-sc-5awhqw-2 bNAbEZ"
          >
            Tiger King
          </p>
          <p
            class="StyledCard__Text-sc-5awhqw-4 bLrHhJ"
          >
            Tiger King description
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    class="StyledCard__Container-sc-5awhqw-1 FaQVJ"
  >
    <h1
      class="StyledCard__Title-sc-5awhqw-0 fFvuDs"
    >
      Feel Good
    </h1>
    <div
      class="StyledCard__Entities-sc-5awhqw-12 jAzA-dB"
    >
      <div
        class="StyledCard__Item-sc-5awhqw-13 hIuTEY"
      >
        <img
          class="StyledCard__Poster-sc-5awhqw-14 hGbFEx"
          src="/assets/images/series/feel-good/juno/small.jpg"
        />
        <div
          class="StyledCard__Meta-sc-5awhqw-11 cwrXet"
        >
          <p
            class="StyledCard__SubTitle-sc-5awhqw-2 bNAbEZ"
          >
            Juno
          </p>
          <p
            class="StyledCard__Text-sc-5awhqw-4 bLrHhJ"
          >
            Juno description
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
